

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Taxonomic annotation of reads and assembled contigs &#8212; Metagenomics Analysis with QIIME 2</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '00-tutorial';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.png" class="logo__image only-light" alt="Metagenomics Analysis with QIIME 2 - Home"/>
    <script>document.write(`<img src="_static/logo.png" class="logo__image only-dark" alt="Metagenomics Analysis with QIIME 2 - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Metagenomics analysis with QIIME 2
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01-sra-data-access.html">Data access from SRA and other resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="02-read-analysis.html">Reads Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="03-contig-analysis.html">Contig Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="04-MAG-analysis.html">MAG analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="80-leiden-workshop.html">Metagenomics with QIIME 2 - Leiden 2024</a></li>

<li class="toctree-l1"><a class="reference internal" href="90-tips.html">Tips for metagenome analysis with QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="91-workflow-diagram.html">Diagram of QIIME 2 metagenomics workflow</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books/issues/new?title=Issue%20on%20page%20%2F00-tutorial.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/00-tutorial.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Taxonomic annotation of reads and assembled contigs</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importing-data">Importing data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#read-taxonomic-annotation-workflow">Read taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-reads-with-kraken2">Classify reads with kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#estimate-abundances-with-bracken">Estimate abundances with bracken</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-taxonomic-annotation-workflow">Contig taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#assemble-reads-with-megahit">Assemble reads with megahit</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-contigs-with-kraken2">Classify contigs with kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#build-presence-absence-feature-table">Build presence/absence feature table</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="taxonomic-annotation-of-reads-and-assembled-contigs">
<h1>Taxonomic annotation of reads and assembled contigs<a class="headerlink" href="#taxonomic-annotation-of-reads-and-assembled-contigs" title="Permalink to this heading">#</a></h1>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Metagenomics analysis with QIIME 2 is in alpha release.
This means that results you generate should be considered preliminary, and NOT PUBLICATION QUALITY.
Additionally, interfaces are subject to change, and those changes may be backward incompatible (meaning that a command or file that works in one version of the QIIME 2 Shotgun Metagenomics distribution may not work in the next version of that distribution).</p>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Filtering of human and other host-associated reads from metagenomics data is an active area of research,
and for human hosts in particular updated approaches need to be developed and validated that filter based on
the <a class="reference external" href="https://humanpangenome.org/">human pangenome</a>, rather than a single human genome sequence. We don’t have
a recommendation on a tool for this right now, but some are in development. Recent publications (<a class="reference external" href="https://genome.cshlp.org/content/29/6/954.long">1</a>, <a class="reference external" href="https://doi.org/10.1038/s41564-023-01381-3">2</a>)
suggest that <strong>personally identifying information can persist in human-derived metagenome data,
even when following current best practices for host read removal</strong>. QIIME 2 has not solved this problem.</p>
<p>All of that said, you can use <a class="reference external" href="https://docs.qiime2.org/2023.9/plugins/available/quality-control/filter-reads/"><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">quality-control</span> <span class="pre">filter-reads</span></code></a>
to filter reads that hit the host genome based on alignment to that genome. <a class="reference external" href="https://docs.qiime2.org/2023.9/plugins/available/quality-control/bowtie2-build/"><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">quality-control</span> <span class="pre">bowtie2-build</span></code></a>
can help you build a bowtie2 database for use with <code class="docutils literal notranslate"><span class="pre">filter-reads</span></code>.</p>
<p>We also recommend filtering again following taxonomy assignment using <a class="reference external" href="https://docs.qiime2.org/2023.9/plugins/available/taxa/filter-seqs/"><code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">taxa</span> <span class="pre">filter-seqs</span></code></a>
to only include features that are assigned to a bacterial phylum, for example (see <a class="reference external" href="https://docs.qiime2.org/2023.9/tutorials/filtering/#taxonomy-based-filtering-of-tables-and-sequences">here</a>).
Note that this is not perfect due to the issues pointed out <a class="reference external" href="https://genome.cshlp.org/content/29/6/954.long">in the first publication referenced above</a>.</p>
<p>In summary, you can do host read filtering with QIIME 2, but the current best practices
in general (not only in QIIME 2) are known to be insufficient. Developing and validating
new best practices is ongoing work.</p>
</div>
<section id="importing-data">
<h2>Importing data<a class="headerlink" href="#importing-data" title="Permalink to this heading">#</a></h2>
<p>Using code from <a class="reference external" href="https://github.com/gregcaporaso/fq-manifestor">this repo</a>, create a manifest.</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">fq_manifestor</span> <span class="kn">import</span> <span class="n">fq_manifestor</span>
<span class="n">fq_manifestor</span><span class="p">(</span>
    <span class="s2">&quot;fastq-dir/&quot;</span><span class="p">,</span>
    <span class="s2">&quot;manifest.tsv&quot;</span><span class="p">,</span>
    <span class="n">f_read_pattern</span><span class="o">=</span><span class="s1">&#39;paired_1&#39;</span><span class="p">,</span>
    <span class="n">r_read_pattern</span><span class="o">=</span><span class="s1">&#39;paired_2&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>tools<span class="w"> </span>import<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--type<span class="w"> </span><span class="s1">&#39;SampleData[PairedEndSequencesWithQuality]&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--input-path<span class="w"> </span>manifest.tsv<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--output-path<span class="w"> </span>demux.qza<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--input-format<span class="w"> </span>PairedEndFastqManifestPhred33V2
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>demux<span class="w"> </span>summarize<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-data<span class="w"> </span>demux.qza<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span>demux.qzv
</pre></div>
</div>
<p>Generate a table of the sequence counts per sample so we can use that as metadata later (e.g., we should always color points in PCoA by read count, to get a feel for whether that is impacting our results).</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>demux<span class="w"> </span>tabulate-read-counts<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-sequences<span class="w"> </span>demux.qza<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-counts<span class="w"> </span>demultiplexed-read-counts.qza
</pre></div>
</div>
</section>
<section id="read-taxonomic-annotation-workflow">
<h2>Read taxonomic annotation workflow<a class="headerlink" href="#read-taxonomic-annotation-workflow" title="Permalink to this heading">#</a></h2>
<section id="classify-reads-with-kraken2">
<h3>Classify reads with kraken2<a class="headerlink" href="#classify-reads-with-kraken2" title="Permalink to this heading">#</a></h3>
<ul class="simple">
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-confidence</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-minimum-base-quality</span></code> are deviations from kraken’s defaults.</p></li>
<li><p>The database used here is the <code class="docutils literal notranslate"><span class="pre">PlusPF</span></code> database, defined <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">here</a>.</p></li>
<li><p>The abbreviations in my <code class="docutils literal notranslate"><span class="pre">output-dir</span></code> are the database (<code class="docutils literal notranslate"><span class="pre">k2pf</span></code>), and shorthand for the values I set for confidence (<code class="docutils literal notranslate"><span class="pre">c60</span></code>) and minimum base quality (<code class="docutils literal notranslate"><span class="pre">mbq20</span></code>), respectively.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>classify-kraken2<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--i-seqs<span class="w"> </span>demux.qza<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--i-kraken2-db<span class="w"> </span>/projects/microbiome/biological-reference-data/2023.06.05-k2-plus-pf-kraken-db.qza<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--p-threads<span class="w"> </span><span class="m">40</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--p-confidence<span class="w"> </span><span class="m">0</span>.6<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--p-minimum-base-quality<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--output-dir<span class="w"> </span>kraken-k2pf-c60-mbq20-reads<span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--p-report-minimizer-data
</pre></div>
</div>
</section>
<section id="estimate-abundances-with-bracken">
<h3>Estimate abundances with bracken<a class="headerlink" href="#estimate-abundances-with-bracken" title="Permalink to this heading">#</a></h3>
<ul class="simple">
<li><p>As far as I know, read length should be the single-end read length for the data (look in the <code class="docutils literal notranslate"><span class="pre">demux.qzv</span></code> generated above, if you don’t know this offhand).</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>estimate-bracken<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-bracken-db<span class="w"> </span>/projects/microbiome/biological-reference-data/2023.06.05-k2-plus-pf-bracken-db.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-read-len<span class="w"> </span><span class="m">150</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-kraken-reports<span class="w"> </span>kraken-k2pf-c60-mbq20-reads/reports.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-reports<span class="w"> </span>kraken-k2pf-c60-mbq20-reads/bracken-reports.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-taxonomy<span class="w"> </span>kraken-k2pf-c60-mbq20-reads/taxonomy-bracken.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-table<span class="w"> </span>kraken-k2pf-c60-mbq20-reads/table-bracken.qza
</pre></div>
</div>
</section>
</section>
<section id="contig-taxonomic-annotation-workflow">
<h2>Contig taxonomic annotation workflow<a class="headerlink" href="#contig-taxonomic-annotation-workflow" title="Permalink to this heading">#</a></h2>
<section id="assemble-reads-with-megahit">
<h3>Assemble reads with megahit<a class="headerlink" href="#assemble-reads-with-megahit" title="Permalink to this heading">#</a></h3>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>assembly<span class="w"> </span>assemble-megahit<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-seqs<span class="w"> </span>demux.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-cpu-threads<span class="w"> </span><span class="m">40</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-contigs<span class="w"> </span>megahit-contigs.qza
</pre></div>
</div>
</section>
<section id="classify-contigs-with-kraken2">
<h3>Classify contigs with kraken2<a class="headerlink" href="#classify-contigs-with-kraken2" title="Permalink to this heading">#</a></h3>
<ul class="simple">
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-confidence</span></code> value is a deviation from kraken’s default.</p></li>
<li><p>The database used here is the <code class="docutils literal notranslate"><span class="pre">PlusPF</span></code> database, defined <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">here</a>.</p></li>
<li><p>The abbreviations in my <code class="docutils literal notranslate"><span class="pre">output-dir</span></code> are the database (<code class="docutils literal notranslate"><span class="pre">k2pf</span></code>), and shorthand for the value I set for confidence (<code class="docutils literal notranslate"><span class="pre">c60</span></code>).</p></li>
<li><p>Decreasing the confidence should increase sensitivity at the expense of specificity.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>classify-kraken2<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-seqs<span class="w"> </span>megahit/megahit-contigs.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-kraken2-db<span class="w"> </span>/projects/microbiome/biological-reference-data/2023.06.05-k2-plus-pf-kraken-db.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">40</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-confidence<span class="w"> </span><span class="m">0</span>.6<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output-dir<span class="w"> </span>kraken-k2pf-c60-contigs/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-report-minimizer-data<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
</section>
<section id="build-presence-absence-feature-table">
<h3>Build presence/absence feature table<a class="headerlink" href="#build-presence-absence-feature-table" title="Permalink to this heading">#</a></h3>
<ul class="simple">
<li><p>The abbreviations in my output file names are shorthand for the value I set for coverage threshold (<code class="docutils literal notranslate"><span class="pre">ct01</span></code>).</p></li>
<li><p>Decreasing the coverage threshold should increase sensitivity at the expense of specificity.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>kraken2-to-features<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-reports<span class="w"> </span>kraken-k2pf-c60-contigs/reports.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-table<span class="w"> </span>kraken-k2pf-c60-contigs/pa-table-ct01.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-taxonomy<span class="w"> </span>kraken-k2pf-c60-contigs/taxonomy-ct01.qza<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-coverage-threshold<span class="w"> </span><span class="m">0</span>.01
</pre></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#importing-data">Importing data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#read-taxonomic-annotation-workflow">Read taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-reads-with-kraken2">Classify reads with kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#estimate-abundances-with-bracken">Estimate abundances with bracken</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-taxonomic-annotation-workflow">Contig taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#assemble-reads-with-megahit">Assemble reads with megahit</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-contigs-with-kraken2">Classify contigs with kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#build-presence-absence-feature-table">Build presence/absence feature table</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Bokulich and Caporaso Labs
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>