

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Contig Analysis &#8212; Metagenomics Analysis with QIIME 2</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '03-contig-analysis';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="MAG analysis" href="04-MAG-analysis.html" />
    <link rel="prev" title="Reads Analysis" href="02-read-analysis.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.png" class="logo__image only-light" alt="Metagenomics Analysis with QIIME 2 - Home"/>
    <script>document.write(`<img src="_static/logo.png" class="logo__image only-dark" alt="Metagenomics Analysis with QIIME 2 - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Metagenomics analysis with QIIME 2
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01-sra-data-access.html">Data access from SRA and other resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="02-read-analysis.html">Reads Analysis</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Contig Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="04-MAG-analysis.html">MAG analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="90-tips.html">Tips for metagenome analysis with QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="91-workflow-diagram.html">Diagram of QIIME 2 metagenomics workflow</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books/issues/new?title=Issue%20on%20page%20%2F03-contig-analysis.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/03-contig-analysis.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Contig Analysis</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#assemble-reads-into-contigs-with-megahit">Assemble Reads into Contigs with MEGAHIT</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-qc-with-quast">Contig QC with QUAST</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-taxonomic-annotation-workflow">Contig Taxonomic Annotation Workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-contigs-with-kraken2">Classify Contigs with Kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#presence-absence-feature-table-creation">Presence/Absence Feature Table Creation</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#filtering-feature-table-and-normalization">Filtering Feature Table and Normalization</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#alpha-diversity-on-presence-absence-feature-table">Alpha Diversity on Presence/Absence Feature Table</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linear-mixed-effects">Linear Mixed Effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#beta-diversity">Beta Diversity</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#jaccard-distance-matrix-pcoa-creation">Jaccard Distance Matrix PCoA creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#emperor-plot-creation">Emperor Plot Creation</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxa-bar-creation">Taxa-bar Creation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-functional-annotation-workflow">Contig functional annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eggnog-search-using-diamond-aligner">EggNOG search using diamond aligner</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-diversity">Gene diversity</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Jaccard Distance Matrix PCoA creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Emperor Plot Creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#annotate-orthologs-against-eggnog-database">Annotate orthologs against eggNOG database</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="contig-analysis">
<h1>Contig Analysis<a class="headerlink" href="#contig-analysis" title="Permalink to this heading">#</a></h1>
<p>This section of the tutorial focuses on obtaining and analyzing contigs, which are contiguous sequences of DNA assembled from short reads obtained through sequencing techniques. Contigs are crucial in genome assembly and analysis.</p>
<section id="assemble-reads-into-contigs-with-megahit">
<h2>Assemble Reads into Contigs with MEGAHIT<a class="headerlink" href="#assemble-reads-into-contigs-with-megahit" title="Permalink to this heading">#</a></h2>
<p>The first step in recovering metagenome-assembled genomes (MAGs) is genome assembly itself. There are many genome assemblers available, two of which you can use through our QIIME 2 plugin - here, we will use MEGAHIT. MEGAHIT takes short DNA sequencing reads, constructs a simplified De Bruijn graph, and generates longer contiguous sequences called contigs, providing valuable genetic information for the next steps of our analysis.</p>
<ul class="simple">
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-num-partition</span></code> specifies the number of partitions to split the dataset into for parallel processing during assembly.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-presets</span></code> specifies the preset mode for MEGAHIT. In this case, it’s set to “meta-sensitive” for metagenomic data.</p></li>
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-cpu-threads</span></code> specifies the number of CPU threads to use during assembly.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>assembly<span class="w"> </span>assemble-megahit<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-seqs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:reads_no_host&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-presets<span class="w"> </span><span class="s2">&quot;meta-sensitive&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-cpu-threads<span class="w"> </span><span class="m">64</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-partitions<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<p>Alternatively, you can also use <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">assembly</span> <span class="pre">assemble-spades</span></code> to assemble contigs with SPAdes.</p>
</section>
<section id="contig-qc-with-quast">
<h2>Contig QC with QUAST<a class="headerlink" href="#contig-qc-with-quast" title="Permalink to this heading">#</a></h2>
<p>Once the reads are assembled into contigs, we can use QUAST to evaluate the quality of our assembly. There are many metrics which can be used for that purpose but here we will focus on the two most popular metrics:</p>
<ul class="simple">
<li><p><strong>N50</strong>: represents the contiguity of a genome assembly. It’s defined as the length of the contig (or scaffold) at which 50% of the entire genome is covered by contigs of that length or longer - the higher this number, the better.</p></li>
<li><p><strong>L50</strong>: represents the number of contigs required to cover 50% of the genome’s total length - the smaller this number, the better.</p></li>
</ul>
<p>In addition to calculating generic statistics like N50 and L50, QUAST will try to identify potential genomes from which the analyzed contigs originated. Alternatively, we can provide it with a set of reference genomes we would like it to run the analysis against using <code class="docutils literal notranslate"><span class="pre">--i-references</span></code>.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>assembly<span class="w"> </span>evaluate-contigs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">128</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-memory-efficient<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/contigs.qzv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
</section>
<section id="contig-taxonomic-annotation-workflow">
<h2>Contig Taxonomic Annotation Workflow<a class="headerlink" href="#contig-taxonomic-annotation-workflow" title="Permalink to this heading">#</a></h2>
<p>Now we are ready to perform taxonomic classification of our contigs.</p>
<section id="classify-contigs-with-kraken2">
<h3>Classify Contigs with Kraken2<a class="headerlink" href="#classify-contigs-with-kraken2" title="Permalink to this heading">#</a></h3>
<p>Here, we are focusing on <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0">Kraken 2</a> - one of the most popular taxonomic classifiers for metagenomic data.
Kraken 2 requires a pre-built database, which we have already built on the <code class="docutils literal notranslate"><span class="pre">01-sra-data-access</span></code> section of this tutorial, so that it can compare the analyzed genomes to a reference. In this example, we are using the <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">Standard database</a>, which is a database built using all archaeal, bacterial, viral, plasmid and human sequences found in the NCBI’s RefSeq database. Since Kraken 2 classification is based on comparing k-mer profiles, this database contains pre-calculated k-mer profiles for all the genomes listed earlier and stored in the so-called “hash tables” - data structres optimized for efficient data storage and retrieval. Alternatively, you can also use <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">moshpit</span> <span class="pre">classify-kaiju</span></code> to classify your contigs with <a class="reference external" href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a>.</p>
<ul class="simple">
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-confidence</span></code> and <code class="docutils literal notranslate"><span class="pre">--p-minimum-base-quality</span></code> are deviations from kraken’s defaults.</p></li>
<li><p>The database used here is the <code class="docutils literal notranslate"><span class="pre">Standard</span></code> database, defined <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">here</a>.</p></li>
<li><p>The abbreviations in my <code class="docutils literal notranslate"><span class="pre">output-dir</span></code> are the database (<code class="docutils literal notranslate"><span class="pre">k2pf</span></code>), and shorthand for the values I set for confidence (<code class="docutils literal notranslate"><span class="pre">c60</span></code>) and minimum base quality (<code class="docutils literal notranslate"><span class="pre">mbq20</span></code>), respectively.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>classify-kraken2<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-seqs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:megahit-contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-kraken2-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_standard&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">48</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-confidence<span class="w"> </span><span class="m">0</span>.6<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-minimum-base-quality<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-partitions<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-reports<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_reports_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_hits_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<p>With this previous action we got two new artifacts: FeatureData[Kraken2Report % Properties(‘contigs’)] and FeatureData[Kraken2Output % Properties(‘contigs’)]. The first one contains the Kraken 2 report: a tree-like representation of all the identified taxa. The second one is a list of all contigs with their corresponding identified taxa.</p>
</section>
<section id="presence-absence-feature-table-creation">
<h3>Presence/Absence Feature Table Creation<a class="headerlink" href="#presence-absence-feature-table-creation" title="Permalink to this heading">#</a></h3>
<p>A natural next step would now be to estimate the relative frequencies of those taxa in our samples, however this is not yet possible to do on contigs with QIIME 2 (coming soon though!) Therefore, to convert those into a more QIIME-like taxonomy, run the following action:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>kraken2-to-features<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-reports<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_reports_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-taxonomy<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_taxonomy_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--verbose
</pre></div>
</div>
</section>
</section>
<section id="filtering-feature-table-and-normalization">
<h2>Filtering Feature Table and Normalization<a class="headerlink" href="#filtering-feature-table-and-normalization" title="Permalink to this heading">#</a></h2>
<p>Once we have feature table, this is becomes alot more similar to the amplicon workflow of QIIME 2.</p>
<p>In this tutorial, we’re going to work specifically with samples that were included in the autoFMT randomized trial. Many of these subjects dropped out before randomization (placing the subject into FMT group or Control group) and therefore do not have a value in the <code class="docutils literal notranslate"><span class="pre">autoFmtGroup</span></code>.</p>
<p>We need to filter our feature table to contain samples that were in the autoFMT study by filtering out any samples that are null in the metadata column autoFmtGroup.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>feature-table<span class="w"> </span>filter-samples<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-where<span class="w"> </span><span class="s1">&#39;autoFmtGroup IS NOT NULL&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-filtered-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_autofmt_feature_table_contigs&quot;</span>
</pre></div>
</div>
</section>
<section id="alpha-diversity-on-presence-absence-feature-table">
<h2>Alpha Diversity on Presence/Absence Feature Table<a class="headerlink" href="#alpha-diversity-on-presence-absence-feature-table" title="Permalink to this heading">#</a></h2>
<p>First we’ll look for general patterns, by comparing different categorical groupings of samples to see if there is some relationship to richness.</p>
<p>To start with, we’ll generate an ‘observed features’ vector from our presence/absence feature table:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>diversity<span class="w"> </span>alpha<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_autofmt_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-metric<span class="w"> </span><span class="s2">&quot;observed_features&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-alpha-diversity<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:obs_features_autofmt_contigs&quot;</span>
</pre></div>
</div>
<section id="linear-mixed-effects">
<h3>Linear Mixed Effects<a class="headerlink" href="#linear-mixed-effects" title="Permalink to this heading">#</a></h3>
<p>In order to manage the repeated measures, we will use a linear
mixed-effects model. In a mixed-effects model, we combine fixed-effects (your typical linear regression coefficients) with random-effects. These random
effects are some (ostensibly random) per-group coefficient which minimizes the
error within that group. In our situation, we would want our random effect to
be the <code class="docutils literal notranslate"><span class="pre">PatientID</span></code> as we can see each subject has a different baseline for
richness (and we have multiple measures for each patient).
By making that a random effect, we can more accurately ascribe associations to
the fixed effects as we treat each sample as a “draw” from a per-group
distribution.</p>
<p>There are several ways to create a linear model with random effects, but we
will be using a <em>random-intercept</em>, which allows for the per-subject intercept
to take on a different average from the population intercept (modeling what we
saw in the group-significance plot above).</p>
<p>First let’s evaluate the general trend of the Bone Marrow transplant.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>qiime<span class="w"> </span>longitudinal<span class="w"> </span>linear-mixed-effects<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:obs_features_autofmt_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-state-column<span class="w"> </span>DayRelativeToNearestHCT<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-individual-id-column<span class="w"> </span>PatientID<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-metric<span class="w"> </span>observed_features<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/lme_obs_features_HCT_contigs.qzv&quot;</span>
</pre></div>
</div>
<p>We may also be interested in the effect of the auto fecal microbiota transplant. It should be known that these are generally correlated, so choosing one model over the other will require external knowledge.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>qiime<span class="w"> </span>longitudinal<span class="w"> </span>linear-mixed-effects<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:obs_features_autofmt_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-state-column<span class="w"> </span>day-relative-to-fmt<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-individual-id-column<span class="w"> </span>PatientID<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--p-metric<span class="w"> </span>observed_features<span class="w"> </span><span class="se">\</span>
<span class="w">   </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/lme_obs_features_FMT_contigs.qzv&quot;</span>
</pre></div>
</div>
<p>We also see a downward trend from the FMT. Since the goal of the FMT was to
ameliorate the impact of the bone marrow transplant protocol (which involves
an extreme course of antibiotics) on gut health, and the timing of the FMT is
related to the timing of the marrow transplant, we might deduce that the
negative coefficient is primarily related to the bone marrow transplant
procedure. (We can’t prove this with statistics alone however, in this case,
we are using abductive reasoning).</p>
<p>Looking at the log-likelihood, we also note that the HCT result is slightly
better than the FMT in accounting for the loss of richness. But only slightly,
if we were to perform model testing it may not prove significant.</p>
<p>In any case, we can ask a more targeted question to identify if the FMT was
useful in recovering richness.</p>
<p>By adding the <code class="docutils literal notranslate"><span class="pre">autoFmtGroup</span></code> to our linear model, we can see if there
are different slopes for the two groups, based on an <em>interaction term</em>.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>longitudinal<span class="w"> </span>linear-mixed-effects<span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:obs_features_autofmt_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-state-column<span class="w"> </span>day-relative-to-fmt<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-group-columns<span class="w"> </span>autoFmtGroup<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-individual-id-column<span class="w"> </span>PatientID<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-metric<span class="w"> </span>observed_features<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/lme_obs_features_treatmentVScontrol_contigs.qzv&quot;</span>
</pre></div>
</div>
</section>
</section>
<section id="beta-diversity">
<h2>Beta Diversity<a class="headerlink" href="#beta-diversity" title="Permalink to this heading">#</a></h2>
<p>Now that we better understand community richness trends, lets look at differences in microbial composition.</p>
<section id="jaccard-distance-matrix-pcoa-creation">
<h3>Jaccard Distance Matrix PCoA creation<a class="headerlink" href="#jaccard-distance-matrix-pcoa-creation" title="Permalink to this heading">#</a></h3>
<p>Let’s first create the Jaccard distance matrix.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>diversity<span class="w"> </span>beta<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_autofmt_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-metric<span class="w"> </span>jaccard<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-distance-matrix<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_autofmt_contigs&quot;</span>
</pre></div>
</div>
<p>Now, let’s generate a PCoA from Jaccard matrix.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>diversity<span class="w"> </span>pcoa<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-distance-matrix<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_autofmt_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-pcoa<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_autofmt_pcoa_contigs&quot;</span>
</pre></div>
</div>
</section>
<section id="emperor-plot-creation">
<h3>Emperor Plot Creation<a class="headerlink" href="#emperor-plot-creation" title="Permalink to this heading">#</a></h3>
<p>Now that we have our Jaccard diversity PCoA, lets visualize it!</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>emperor<span class="w"> </span>plot<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-pcoa<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_autofmt_pcoa_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/jaccard_autofmt_emperor.qzv</span>
</pre></div>
</div>
<p>We can make <code class="docutils literal notranslate"><span class="pre">week-relative-to-fmt</span></code> a custom axis in our PCOA. This allows us to look at changes in microbial composition over the couse of the study.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>emperor<span class="w"> </span>plot<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-pcoa<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_autofmt_pcoa_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-custom-axes<span class="w"> </span>week-relative-to-fmt<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/jaccard_autofmt_emperor_custom.qzv&quot;</span>
</pre></div>
</div>
</section>
</section>
<section id="taxa-bar-creation">
<h2>Taxa-bar Creation<a class="headerlink" href="#taxa-bar-creation" title="Permalink to this heading">#</a></h2>
<p>We will now explore our coting microbial composition by visualizing a taxa bar plot. Note that we are using a <code class="docutils literal notranslate"><span class="pre">FeatureTable[PresenceAbsence]</span></code>, hence we are not talking about relative abundance in this case.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>taxa<span class="w"> </span>barplot<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_autofmt_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-taxonomy<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_taxonomy_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/taxa_bar_plot_autofmt_contigs.qzv&quot;</span>
</pre></div>
</div>
</section>
<section id="contig-functional-annotation-workflow">
<h2>Contig functional annotation workflow<a class="headerlink" href="#contig-functional-annotation-workflow" title="Permalink to this heading">#</a></h2>
<p>Here we will perform functional annotation of contigs to capture gene diversity.</p>
<section id="eggnog-search-using-diamond-aligner">
<h3>EggNOG search using diamond aligner<a class="headerlink" href="#eggnog-search-using-diamond-aligner" title="Permalink to this heading">#</a></h3>
<p>Searches for homologous sequences in the EggNOG database using the Diamond aligner for faster processing.</p>
<ul class="simple">
<li><p>The <code class="docutils literal notranslate"><span class="pre">--p-db-in-memory</span></code>loads the database into memory for faster processing.</p></li>
</ul>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>eggnog-diamond-search<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-sequences<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-diamond-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_diamond_full&quot;</span><span class="se">\</span>
<span class="w">  </span>--p-num-cpus<span class="w"> </span><span class="m">14</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-db-in-memory<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-eggnog-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:diamond_hits_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:diamond_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--verbose
</pre></div>
</div>
</section>
<section id="gene-diversity">
<h3>Gene diversity<a class="headerlink" href="#gene-diversity" title="Permalink to this heading">#</a></h3>
<p>Now, let’s have a look at the gene composition of our community</p>
</section>
<section id="id1">
<h3>Jaccard Distance Matrix PCoA creation<a class="headerlink" href="#id1" title="Permalink to this heading">#</a></h3>
<p>We will again start by calculating our Jaccard beta-diversity matrix</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>diversity<span class="w"> </span>beta<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:diamond_feature_table_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-metric<span class="w"> </span>jaccard<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-distance-matrix<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_distance_matrix_contigs&quot;</span>
</pre></div>
</div>
<p>Then, we will generate out PCoA from Jaccard matrix</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>diversity<span class="w"> </span>pcoa<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-distance-matrix<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_distance_matrix_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-pcoa<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_distance_matrix_pcoa_contigs&quot;</span>
</pre></div>
</div>
</section>
<section id="id2">
<h3>Emperor Plot Creation<a class="headerlink" href="#id2" title="Permalink to this heading">#</a></h3>
<p>Now that we have our Jaccard diversity PCoA, lets visualize it!</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>emperor<span class="w"> </span>plot<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-pcoa<span class="w">  </span><span class="s2">&quot;./moshpit_tutorial/cache:jaccard_distance_matrix_pcoa_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--m-metadata-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/metadata.tsv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/jaccard_distance_matrix_pcoa_contigs.qzv&quot;</span>
</pre></div>
</div>
</section>
<section id="annotate-orthologs-against-eggnog-database">
<h3>Annotate orthologs against eggNOG database<a class="headerlink" href="#annotate-orthologs-against-eggnog-database" title="Permalink to this heading">#</a></h3>
<p>Lastly, let’s annotate our contigs with functional information from the EggNOG database.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>eggnog-annotate<span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-eggnog-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-eggnog-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_annot_full&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--o-ortholog-annotations<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_annotated_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--verbose
</pre></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="02-read-analysis.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Reads Analysis</p>
      </div>
    </a>
    <a class="right-next"
       href="04-MAG-analysis.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">MAG analysis</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#assemble-reads-into-contigs-with-megahit">Assemble Reads into Contigs with MEGAHIT</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-qc-with-quast">Contig QC with QUAST</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-taxonomic-annotation-workflow">Contig Taxonomic Annotation Workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#classify-contigs-with-kraken2">Classify Contigs with Kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#presence-absence-feature-table-creation">Presence/Absence Feature Table Creation</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#filtering-feature-table-and-normalization">Filtering Feature Table and Normalization</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#alpha-diversity-on-presence-absence-feature-table">Alpha Diversity on Presence/Absence Feature Table</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#linear-mixed-effects">Linear Mixed Effects</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#beta-diversity">Beta Diversity</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#jaccard-distance-matrix-pcoa-creation">Jaccard Distance Matrix PCoA creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#emperor-plot-creation">Emperor Plot Creation</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#taxa-bar-creation">Taxa-bar Creation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#contig-functional-annotation-workflow">Contig functional annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eggnog-search-using-diamond-aligner">EggNOG search using diamond aligner</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-diversity">Gene diversity</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Jaccard Distance Matrix PCoA creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Emperor Plot Creation</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#annotate-orthologs-against-eggnog-database">Annotate orthologs against eggNOG database</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Bokulich and Caporaso Labs
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>