

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>MAG analysis &#8212; Metagenomics Analysis with QIIME 2</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = '04-MAG-analysis';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Metagenomics with QIIME 2 - Leiden 2024" href="80-leiden-workshop.html" />
    <link rel="prev" title="Contig Analysis" href="03-contig-analysis.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.png" class="logo__image only-light" alt="Metagenomics Analysis with QIIME 2 - Home"/>
    <script>document.write(`<img src="_static/logo.png" class="logo__image only-dark" alt="Metagenomics Analysis with QIIME 2 - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Metagenomics analysis with QIIME 2
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="01-sra-data-access.html">Data access from SRA and other resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="02-read-analysis.html">Reads Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="03-contig-analysis.html">Contig Analysis</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">MAG analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="80-leiden-workshop.html">Metagenomics with QIIME 2 - Leiden 2024</a></li>

<li class="toctree-l1"><a class="reference internal" href="90-tips.html">Tips for metagenome analysis with QIIME 2</a></li>
<li class="toctree-l1"><a class="reference internal" href="91-workflow-diagram.html">Diagram of QIIME 2 metagenomics workflow</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/caporaso-lab/q2-books/issues/new?title=Issue%20on%20page%20%2F04-MAG-analysis.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/04-MAG-analysis.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>MAG analysis</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bin-contigs-into-mags">Bin contigs into MAGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#read-mapping">Read mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#binning">Binning</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-dereplication-optional">MAGs dereplication (optional)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-qc-with-busco">MAGs QC with BUSCO</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-taxonomic-annotation-workflow">MAGs taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-classify-with-kraken2">MAGs classify with Kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#build-taxonomy-table">Build taxonomy table</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-functional-annotation-workflow">MAGs functional annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eggnog-search-using-diamond-aligner">EggNOG search using diamond aligner</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#annotate-orthologs-against-eggnog-database">Annotate orthologs against eggNOG database</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="mag-analysis">
<h1>MAG analysis<a class="headerlink" href="#mag-analysis" title="Permalink to this heading">#</a></h1>
<p>This section of the tutorial focuses on analyzing metagenome-assembled genomes (MAGs), which are reconstructed genomes derived from metagenomic data.</p>
<section id="bin-contigs-into-mags">
<h2>Bin contigs into MAGs<a class="headerlink" href="#bin-contigs-into-mags" title="Permalink to this heading">#</a></h2>
<p>Let’s start binning our contigs into MAGs using various tools and methodologies!</p>
<section id="read-mapping">
<h3>Read mapping<a class="headerlink" href="#read-mapping" title="Permalink to this heading">#</a></h3>
<p>We first need to index the contigs obtained in the assembly step and map the original reads to those contigs using that index. This read mapping can then be used by the contig binner to figure out which contigs originated from the same genome and put those together.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>assembly<span class="w"> </span>index-contigs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-seed<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">64</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-verbose<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-partitions<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-index<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs_index&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>assembly<span class="w"> </span>map-reads-to-contigs<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-indexed-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs_index&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-reads<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:reads_no_host&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-seed<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">64</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-partitions<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-alignment-map<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:reads_to_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
</section>
<section id="binning">
<h3>Binning<a class="headerlink" href="#binning" title="Permalink to this heading">#</a></h3>
<p>Finally, we are ready to perform contig binning. This process involves categorizing contigs into distinct bins or groups based on their likely origin from different microbial species or strains within a mixed community. Here, we will use the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6662567/">MetaBAT 2</a> tool, which uses tetranucleotide frequency together with abundance (coverage) information to assign contigs to individual bins.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>bin-contigs-metabat<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-alignment-maps<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:reads_to_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-seed<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-threads<span class="w"> </span><span class="m">128</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-verbose<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-mags<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-contig-map<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:contig_map&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-unbinned-contigs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:unbinned_contigs&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<p>This tep generated a couple artifacts:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">mags.qza</span></code>: these are our actual MAGS, per sample.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">contig-map.qza</span></code>: this is a mapping between MAG IDs and IDs of contigs which belong to a given MAG.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">unbinned-contigs.qza</span></code>: these are all the contigs that could not be assign to any particular MAG.
From now on, we will focus on the mags.qza.</p></li>
</ul>
</section>
</section>
<section id="mags-dereplication-optional">
<h2>MAGs dereplication (optional)<a class="headerlink" href="#mags-dereplication-optional" title="Permalink to this heading">#</a></h2>
<p>Dereplication involves removing duplicate or nearly identical MAGs to reduce redundancy and improve downstream analyses. To dereplicate our MAGs, we will:</p>
<ol class="arabic simple">
<li><p>compute hash sketches of every genome using <a class="reference external" href="https://sourmash.readthedocs.io/en/latest/">sourmash</a> - you can think of those sketches as tiny representations of our genomes (sourmash compresses a lot of information into much smaller space).</p></li>
<li><p>compare all of those sketches (genomes) to one another to generate a matrix of pairwise distances between our MAGs</p></li>
<li><p>dereplicate the genomes using the distance matrix and a fixed similarity threshold: the last action will simply choose the longest genome from all of the genomes belonging to the same cluster, given a similarity threshold.</p></li>
</ol>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>sourmash<span class="w"> </span>compute<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-sequence-file<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-ksizes<span class="w"> </span><span class="m">35</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-scaled<span class="w"> </span><span class="m">10</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-min-hash-signature<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_minhash&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>sourmash<span class="w"> </span>compare<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-min-hash-signature<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_minhash&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-ksize<span class="w"> </span><span class="m">35</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-compare-output<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_dist_matrix&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>dereplicate-mags<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-mags<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-distance-matrix<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_dist_matrix&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threshold<span class="w"> </span><span class="m">0</span>.99<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-dereplicated-mags<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_derep&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-feature-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags_ft&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
</section>
<section id="mags-qc-with-busco">
<h2>MAGs QC with BUSCO<a class="headerlink" href="#mags-qc-with-busco" title="Permalink to this heading">#</a></h2>
<p>BUSCO is used here to assess the completeness and quality of MAGs by searching for single-copy orthologous genes within the genomes.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>evaluate-busco<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-bins<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-lineage-dataset<span class="w"> </span>bacteria_odb10<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-cpu<span class="w"> </span><span class="m">196</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-visualization<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/results/mags.qzv&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
</section>
<section id="mags-taxonomic-annotation-workflow">
<h2>MAGs taxonomic annotation workflow<a class="headerlink" href="#mags-taxonomic-annotation-workflow" title="Permalink to this heading">#</a></h2>
<p>This workflow focuses on annotating MAGs with taxonomic information using Kraken2, a tool for taxonomic classification. In this tutorial we perfom taxonomic and functional annotation on dereplicated MAGs.</p>
<section id="mags-classify-with-kraken2">
<h3>MAGs classify with Kraken2<a class="headerlink" href="#mags-classify-with-kraken2" title="Permalink to this heading">#</a></h3>
<p>MAGs are classified taxonomically using Kraken2, with parameters set for confidence threshold and minimum base quality.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>classify-kraken2<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-seqs<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--i-kraken2-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_standard&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-threads<span class="w"> </span><span class="m">48</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-confidence<span class="w"> </span><span class="m">0</span>.6<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-minimum-base-quality<span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--p-num-partitions<span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-reports<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_reports_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--o-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_hit_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--verbose
</pre></div>
</div>
<p>Alternatively, you can also use <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">moshpit</span> <span class="pre">classify-kaiju</span></code> to classify your contigs with Kaiju.</p>
</section>
<section id="build-taxonomy-table">
<h3>Build taxonomy table<a class="headerlink" href="#build-taxonomy-table" title="Permalink to this heading">#</a></h3>
<p>A taxonomy table is constructed based on the classification results, providing taxonomic assignments for each MAG.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>kraken2-to-mag-features<span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-reports<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_reports_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_hits_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--o-taxonomy<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:kraken_taxonomy_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--verbose
</pre></div>
</div>
</section>
</section>
<section id="mags-functional-annotation-workflow">
<h2>MAGs functional annotation workflow<a class="headerlink" href="#mags-functional-annotation-workflow" title="Permalink to this heading">#</a></h2>
<p>This workflow involves annotating MAGs with functional information, particularly identifying functional elements like genes and their annotations.</p>
<section id="eggnog-search-using-diamond-aligner">
<h3>EggNOG search using diamond aligner<a class="headerlink" href="#eggnog-search-using-diamond-aligner" title="Permalink to this heading">#</a></h3>
<p>MAGs are searched against the EggNOG database using the Diamond aligner to identify functional annotations.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>eggnog-diamond-search<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-sequences<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--i-diamond-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_diamond_full&quot;</span><span class="se">\</span>
<span class="w">  </span>--p-num-cpus<span class="w"> </span><span class="m">14</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--p-db-in-memory<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-eggnog-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:diamond_hits_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--o-table<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:diamond_feature_table_derep_mags \</span>
<span class="s2">  --verbose</span>
</pre></div>
</div>
</section>
<section id="annotate-orthologs-against-eggnog-database">
<h3>Annotate orthologs against eggNOG database<a class="headerlink" href="#annotate-orthologs-against-eggnog-database" title="Permalink to this heading">#</a></h3>
<p>Orthologs in MAGs are annotated against the EggNOG database, providing functional insights into the genes and gene products present in the MAGs.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>qiime<span class="w"> </span>moshpit<span class="w"> </span>eggnog-annotate<span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-eggnog-hits<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--i-eggnog-db<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_annot_full&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--o-ortholog-annotations<span class="w"> </span><span class="s2">&quot;./moshpit_tutorial/cache:eggnog_annotated_derep_mags&quot;</span><span class="w"> </span><span class="se">\</span>
<span class="w"> </span>--verbose
</pre></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="03-contig-analysis.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Contig Analysis</p>
      </div>
    </a>
    <a class="right-next"
       href="80-leiden-workshop.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Metagenomics with QIIME 2 - Leiden 2024</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#bin-contigs-into-mags">Bin contigs into MAGs</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#read-mapping">Read mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#binning">Binning</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-dereplication-optional">MAGs dereplication (optional)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-qc-with-busco">MAGs QC with BUSCO</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-taxonomic-annotation-workflow">MAGs taxonomic annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-classify-with-kraken2">MAGs classify with Kraken2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#build-taxonomy-table">Build taxonomy table</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mags-functional-annotation-workflow">MAGs functional annotation workflow</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#eggnog-search-using-diamond-aligner">EggNOG search using diamond aligner</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#annotate-orthologs-against-eggnog-database">Annotate orthologs against eggNOG database</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Bokulich and Caporaso Labs
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>